client/router.js

```diff
@@ -79,11 +81,15 @@ Router.prototype.path = function(pathDef, fields, queryParams) {
     path += "?" + strQueryParams;
   }
 
+  if(hash) {
+    path += "#" + hash;
+  }
+
   return path;
 };
 
-Router.prototype.go = function(pathDef, fields, queryParams) {
-  var path = this.path(pathDef, fields, queryParams);
+Router.prototype.go = function(pathDef, fields, queryParams, hash) {
+  var path = this.path(pathDef, fields, queryParams, hash);
 
   if (this._current.path !== path) {
```

Comment:
Self note: `this._current.path` does not contain the hash value. So, due to that reason, this equation will be false every time.

If we set the hash to null, theoretically it should remove. But due this reason, path and the `._current.path` is the same. So, we can't remove hash at all.